### ISSUE TITLE
"undefined reference to" while trying to use "libxdo". 

### ISSUE THREAD

**Issue by moscoquera** [jordansissel/xdotool#63]
hi, im trying to use libxdo for a private project.

i'm doing two easy calls to xdo_new and xdo_move_mouse_relative. but when i tried to build my file using this command:

g++  LinuxTest.cpp -lXtst -lX11 -lxdo

but i'm getting:

```
LinuxTest.cpp:(.text+0x34): undefined reference to `xdo_new(char const*)'
LinuxTest.cpp:(.text+0x54): undefined reference to `xdo_move_mouse_relative(xdo const*, int, int)'
```

xdo.h is located at "/usr/local/include/" and libxdo.so is in "/usr/lib/"

**Comment by troiganto** [comment]
Hi, it's been a while since this was posted, but I just stumbled over the same silly mistake and thought that maybe this could get fixed in the main repo one day.

The problem is the name-mangling mechanism that C++ uses to make function names unique. The problem isn't new and you can find some information on [Stack Overflow](http://stackoverflow.com/questions/12066279/using-c-libraries-for-c-programs).
A lot of libraries already contain the `extern "C"` declaration via compiler directives, but libxdo doesn't.

Long story short, to make libxdo work, you have to include its header like that:
```
extern "C" {
#include <xdo.h>
}
```

Maybe we can also turn this into some sort of feature request to get the directive included in xdo.h.
The necessary effort should be minimal.

**Comment by VTiTux** [comment]
Thank you very much.
Could this information added to the documentation?
I wasted one day to trying to build my project with libxdo.
